front_end/panels/ai_chat/ui/settings/advanced/VectorDBSettings.ts

```
@@ -0,0 +1,290 @@
+    vectorDBTestButton.setAttribute('type', 'button');
+    vectorDBTestButton.textContent = i18nString('testVectorDBConnection');
+    vectorDBTestDiv.appendChild(vectorDBTestButton);
+
+    const vectorDBTestStatus = document.createElement('div');
+    vectorDBTestStatus.classList.add('settings-status');
+    vectorDBTestStatus.style.display = 'none';
+    vectorDBTestDiv.appendChild(vectorDBTestStatus);
+
+    // Toggle vector DB config visibility
+    this.vectorDBEnabledCheckbox.addEventListener('change', () => {
+      vectorDBConfigContainer.style.display = this.vectorDBEnabledCheckbox!.checked ? 'block' : 'none';
```

ESLint

[error] Forbidden non-null assertion.

(@typescript-eslint/no-non-null-assertion)